asteroid/data/dampvsep_dataset.py

```diff
@@ -0,0 +1,225 @@
+from pathlib import Path
+from multiprocessing import cpu_count
+
+import torch.utils.data
+import random
+import json
+import librosa
+
+# ignore warning related with
+# https://github.com/librosa/librosa/issues/1015
+# Soundfile can read OGG (vocal) but not M4A (background and mixture)
+import warnings
+
+warnings.filterwarnings("ignore", category=UserWarning)
+
+
+class DAMPVSEPDataset(torch.utils.data.Dataset):
+    """
+    DAMP-VSEP vocal separation dataset
+
+    This dataset utilises one of the two preprocessed versions of DAMP-VSEP
+    from https://github.com/groadabike/DAMP-VSEP-Singles aimed for
+    SINGLE SINGER separation.
+
+    Note: DAMP-VSEP dataset is hosted on Zenodo.
+          https://zenodo.org/record/3553059
+
+    Note 2: There are 2 train set available:
+        1- train_english: Uses all English spoken song.
+            Duets are converted into 2 singles.
+            Totalling 9243 performances and 77Hrs.
+        2- train_singles: Uses all singles performances, discarding all duets.
+            Totalling 20660 performances and 149 hrs.
+    Args:
+        root_path (str): Root path to DAMP-VSEP dataset.
+        task (str): one of ``'enh_vocal'``,``'enh_both'``.
+            * ``'enh_vocal'`` for vocal enhanced.
+            * ``'enh_both'`` for vocal and background separation.
+        split (str):  one of ``'train_english'``, ``'train_singles'``,
+                    ``'valid'`` and ``'test'``.
+                    Default to ``'train_singles'``.
+        samples_per_track (int, optional):
+            Number of samples yielded from each track, can be used to increase
+            dataset size, defaults to ``1``.
```

Comment:
Maybe rename this `ex_per_track`, `samples` usually refer to audio samples in Asteroid and this could be confusing.